jina/peapods/networking.py

```diff
@@ -1,8 +1,341 @@
-from argparse import Namespace
+import ast
+import asyncio
 import ipaddress
+import socket
+from abc import abstractmethod
+from argparse import Namespace
+from threading import Thread
+
+import grpc
 
+from jina.logging.logger import JinaLogger
+from jina.proto import jina_pb2_grpc
+from jina.types.message import Message
 from .. import __default_host__, __docker_host__
-from ..helper import get_public_ip, get_internal_ip
+from ..helper import get_public_ip, get_internal_ip, get_or_reuse_loop
+
+
+class ConnectionList:
+    """
+    Maintains a list of connections and uses round roubin for selecting a connection
+
+    :param port: port to use for the connections
+    """
+
+    def __init__(self, port: int):
+        self.port = port
+        self._connections = []
+        self._address_to_connection_idx = {}
+        self._rr_counter = 0
+
+    def add_connection(self, address: str, connection):
```

Comment:
can u add type hint for `connection`?

Reply:
actually I can't, because connection can be anything from this point of view (grpc stubs, zmq sockets, raw sockets, ..). Also this class does not interact with the object at all, it just stores and returns the object.